dev/com.ibm.ws.jaxrs.2.x.monitor/src/com/ibm/ws/jaxrs/monitor/RESTful_Stats.java

```diff
@@ -0,0 +1,168 @@
+    private String methodName;
+
+    //Following is the stats we are reporting for Resource Methods.
+    private Counter requestCount;
+    private final StatisticsMeter responseTime;
+
+    /**
+     * Constructor.
+     * We will store AppName and Resource Method Name
+     */
+    public RESTful_Stats(String aName, String mName) {
+        setAppName(aName);
+        setMethodName(mName);
+        requestCount = new Counter();
+        requestCount.setDescription("This shows number of requests to a Restful resource method");
+//        requestCount.setUnit("ns");
+        responseTime = new StatisticsMeter();
+        responseTime.setDescription("Cumulative Response Time (NanoSeconds) for a Restful resource method");
+        responseTime.setUnit("ns");
+    }
+
+    /**
+     * Getter for Application Name.
+     * */
+    @Override
+    public String getAppName() {
+        return this.appName;
+    }
+
+    /**
+     * Getter for Description
+     **/
+    @Override
+    public String getDescription() {
+        return "Report Stats for specified Restful resource method.";
+    }
+
+    /**
+     * Demonstrate Number of Requests, received for this resource method.
+     * 
+     **/
+    @Override
+    public long getRequestCount() {
+        return this.requestCount.getCurrentValue();
+    }
+
+    /**
+     * Reports Average Response Time.
+     * We store Cumulative Time (total service time for specified resource method)
+     * responseTime = CummulativeTime / number of requests.
+     * 
+     * */
+    @Override
+    public double getResponseTime() {
+        return responseTime.getTotal();
+    }
+
+    /**
+     * Getter for Resource Method Name
+     * */
+    @Override
+    public String getMethodName() {
+        return this.methodName;
+    }
+
+    /**
+     * @param appName the appName to set
+     */
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+
+    /**
+     * @param methodName the resource method name to set
+     */
+    public void setMethodName(String methodName) {
+        this.methodName = methodName;
+    }
+
+    /**
+     * @param requestCount the requestCount to set
+     */
+    public void setRequestCount(Counter requestCount) {
+        this.requestCount = requestCount;
+    }
+
+    /**
+     * This will increment request count by i.
+     * Typically i would be 1.
+     * 
+     * @param i
+     */
+    public void incrementCountBy(int i) {
+        this.requestCount.incrementBy(i);
+    }
+
+    /**
+     * updateRT()
+     * 
+     * This is responsible for updating Response Time.
+     * We already have valid serviceStartTime.
+     * Now we should take (current time - serviceStartTime).
+     * Add it to cumulative time.
+     * 
+     * @param elapsed
+     * 
+     * @param l
+     */
+    public void updateRT(long elapsed) {
+        this.responseTime.addDataPoint(elapsed);
+        System.out.println("Jim555, total = " + this.responseTime.getTotal());
```

Comment:
you probably already know this, but this line needs to be deleted.

Reply:
Yup, Yup